packages/react-renderer-demo/src/app/pages/renderer/migration-guide.md

```diff
@@ -0,0 +1,152 @@
+import Grid from '@material-ui/core/Grid'
+import RawComponent from '@docs/raw-component';
+import RouterLink from 'next/link';
+import Link from '@material-ui/core/Link';
+
+import ListOfContents from '../../src/helpers/list-of-contents';
+
+<Grid container item>
+<Grid item xs={12} md={10}>
+
+# Migration guide from 1.x to 2.x
+
+The release of version 2 brings many new features, that as we hope you will find useful and you will like them.
+Also, we took this release as an opportunity to fix mistakes we did during the initial development and to remove some obsolete stuff, that is no longer useful.
+Major themes are using of React hooks, so update your React before you will continue, and giving more freedom to you how to customize the form.
+There are also major enhancements bringing more features to schemas not stored in JavaScript, as the action mapper and validator mapper.
+Thank you for your understanding.
+
+## Details
+
+### General rules
+
+-   All JSON schemas should remain almost the same
+
+### PF4/PF3/MUI Mapper
+
+-   formFieldsMapper was renamed to componentMapper
+-   Components can be imported separately to create only a subset of full component mapper
+
+### Removed 'treshold' option
+
+-   Due to a typo, it was possible to use key treshold as an option in some validators, this option is now disabled. Use only threshold.
+
+### Form Template
+
+-   layoutMapper removed - instead of it use FormTemplate
+-   All props related to form buttons are now removed, use them in you FormTemplate
+    -   showFormControls
```

Comment:
missing props for buttons

Reply:
:question:

Reply:
onCancel, labels, etc are now in formtemplate not renderer props

Reply:
onCancel is still in the renreder. 

buttonsLabels, etc. are mentioned in this list.